app/code/Magento/Directory/Model/Currency/Import/CurrencyConverterApi.php

```diff
@@ -87,9 +97,14 @@ private function convertBatch($data, $currencyFrom, $currenciesTo)
                         $this->_messages[] = __('We can\'t retrieve a rate from %1 for %2.', $url, $to);
                         $data[$currencyFrom][$to] = null;
                     } else {
-                        $data[$currencyFrom][$to] = $this->_numberFormat(
-                            (double)$response[$currencyFrom . '_' . $to]
-                        );
+                        if (isset($response['error']) && $response['error']) {
```

Comment:
It is really nice to have error handling!

Can we check if API key is set in the beginning of the method?
With this we will be able to show error message about missed API key and avoid calls to API in the loop with missed API key.

Reply:
I added it. Please check it again. Thanks.